### PR TITLE
feat: Add delete all done tasks

### PR DESCRIPTION
**Pull request by m1oojv**
## What type of PR is this? (check all applicable)

-   [ ] Refactor
-   [x] Feature
-   [ ] Bug Fix
-   [ ] Optimization
-   [ ] Documentation Update

## Description

-   Add feature to delete all done tasks.

## Related Tickets & Documents

-   Related Issue #114 

## Added/updated tests?

_We encourage you to keep the code coverage percentage at 75% and above for new changes._

-   [x] Yes
-   [ ] No, and this is why: 
-   [ ] No, there are UI changes that I'm unable to cover.
-   [ ] I need help with writing tests

### CODE REVIEW

**Comment by jiakai-17** [comment]
Please do fix the CI before requesting a review.

File: src/main/java/seedu/address/logic/commands/DeleteAllDoneCommand.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        model.updateFilteredTaskList(PREDICATE_SHOW_DONE_TASKS);
+        List<Task> lastShownList = model.getFilteredTaskList();
+        if (lastShownList.isEmpty()) {
+            model.updateFilteredTaskList(PREDICATE_SHOW_ALL_TASKS);
+            throw new CommandException(MESSAGE_NO_DONE_TASKS);
+        }
+        for (int i = lastShownList.size() - 1; i >= 0; i--) {
+            Task task = lastShownList.get(i);
+            model.deleteTask(task);
+        }
```
Comment (freshcabbage123) [inline src/main/java/seedu/address/logic/commands/DeleteAllDoneCommand.java:36]: Would it be better abstraction to implement a deleteAllDoneTasks() method instead inside Model and ModelManager? Similar to how it was implemented for #90

**Review by freshcabbage123 — CHANGES_REQUESTED** [review]
lgtm in general just have some queries

File: src/main/java/seedu/address/logic/commands/DeleteAllDoneCommand.java (hunk 1 above)
Comment (m1oojv) [reply src/main/java/seedu/address/logic/commands/DeleteAllDoneCommand.java:36]: > I think the for loop can be replaced with an enhanced for loop. Something like `for (Task t : lastShownList)`

@jiakai-17 I tried this method, but then it kept on facing with the problem of index error , no element found . and the last marled done element kept on not being able to get deleted.  I suspect the problem is given how the list changes with each iteration of the loop, which is why only deleting from the back is works. 
 which is why i used this loop instead

Comment (m1oojv) [reply src/main/java/seedu/address/logic/commands/DeleteAllDoneCommand.java]: @freshcabbage123 
> Would it be better abstraction to implement a deleteAllDoneTask() method instead inside Model and ModelManager? Similar to how it was implemented for #90 and #77

Moved to Model Manager. thanks

File: src/main/java/seedu/address/model/ModelManager.java [hunk 2]
```diff
@@ -95,6 +95,7 @@ public void setAddressBook(ReadOnlyAddressBook addressBook) {
 
     @Override
     public boolean hasPerson(Person person) {
+        requireNonNull(person);
```
Comment (pra-navi) [inline src/main/java/seedu/address/model/ModelManager.java:98]: This is repeated

**Review by jiakai-17 — APPROVED** [review]

**Review by freshcabbage123 — APPROVED** [review]

**Pull request merged**